### PR TITLE
Allow easier creation of mysql monitoring through hiera

### PR DESCRIPTION
**Pull request by mnellemann**
Would something like this be possible for the mysql plugin ?
I've tested it locally w. my own manifests, but I don't know if it could cause problems elsewhere.

### CODE REVIEW

File: manifests/plugin/mysql.pp [hunk 1]
```diff
@@ -4,6 +4,7 @@
   $ensure           = present,
   $manage_package   = $collectd::manage_package,
   $interval         = undef,
+  $databases        = { },
```
Comment (danzilio) [inline manifests/plugin/mysql.pp]: Can we set this to `false` or `undef` and wrap the call to `create_resources` in an `if` statement? We can use `validate_hash` to ensure that `$databases` is a hash.

Comment (igalic) [inline manifests/plugin/mysql.pp]: create_resources takes all of that away from us
it does nothing on empty hash, and fails horribly if passed anything other than a (valid) hash.

Comment (danzilio) [inline manifests/plugin/mysql.pp]: That's very considerate of `create_resources`! I withdraw my objection.

**Comment by igalic** [comment]
would still be good to have this documented, @mnellemann!

**Comment by jyaworski** [comment]
Hello:

Please rebase and/or squash.

**Pull request closed without merging**